integration_tests/e2e/booking.cy.ts

```diff
@@ -19,26 +19,24 @@ context('Booking', () => {
       keyWorker: 'Alex Evans',
     })
 
-    const premises = premisesFactory.buildList(5)
-    cy.task('stubBookingCreate', { premisesId: 'premisesId', booking })
-    cy.task('stubPremises', premises)
+    const premises = premisesFactory.build()
+    cy.task('stubBookingCreate', { premisesId: premises.id, booking })
+    cy.task('stubBookingGet', { premisesId: premises.id, booking })
+    cy.task('stubSinglePremises', { premisesId: premises.id, booking })
 
     // Given I am signed in
     cy.signIn()
 
     // When I visit the booking page
-    const page = BookingPage.visit()
+    const page = BookingPage.visit(premises.id)
 
     // And I fill in the booking form
     page.completeForm(booking)
     page.clickSubmit()
 
-    // Then I should be redirected to the premises page
-    Page.verifyOnPage(PremisesPage)
-    PremisesPage.shouldShowFlashMessage()
-
-    // And the booking should be created in the API
-    cy.task('verifyBookingCreate', { premisesId: 'premisesId' }).then(requests => {
+    // Then I should be redirected to the confirmation page and the booking should be created in the API
+    Page.verifyOnPage(BookingConfirmation)
```

Comment:
I think we're OK to have the booking confirmation assertion in this part. I get that it makes the test a bit longer, but I think it's an OK tradeoff vs having another spec file to maintain.

Reply:
Sure!